dashboard/app/controllers/home_controller.rb

```diff
@@ -97,38 +99,94 @@ def should_redirect_to_script_overview?
       if script
         script_level = current_user.next_unpassed_progression_level(script)
       end
-
       if script && script_level
-        @top_course = {
+        @homepage_data[:topCourse] = {
           assignableName: data_t_suffix('script.name', script[:name], 'title'),
           lessonName: script_level.stage.localized_title,
           linkToOverview: script_path(script),
           linkToLesson: script_next_path(script, 'next')
         }
       end
+
+      if current_user.teacher?
+        donor_banner_name ||= params[:forceDonorTeacherBanner]
+        show_census_banner = !!(!donor_banner_name && current_user.show_census_teacher_banner?)
+
+        unless current_user.donor_teacher_banner_dismissed
+          donor_banner_name = current_user.donor_teacher_banner_name
+        end
+
+        @homepage_data[:isTeacher] = true
+        @homepage_data[:hocLaunch] = DCDO.get('hoc_launch', CDO.default_hoc_launch)
+        @homepage_data[:joined_sections] = student_sections
+        @homepage_data[:announcement] = DCDO.get('announcement_override', nil)
+        @homepage_data[:hiddenScripts] = current_user.get_hidden_script_ids
+        @homepage_data[:showCensusBanner] = show_census_banner
+        @homepage_data[:donorBannerName] = donor_banner_name
+
+        if show_census_banner
+          teachers_school = current_user.school_info.school
+          school_stats = SchoolStatsByYear.where(school_id: teachers_school.id).order(school_year: :desc).first
+
+          @homepage_data[:censusQuestion] = school_stats.try(:has_high_school_grades?) ? "how_many_20_hours" : "how_many_10_hours"
+          @homepage_data[:currentSchoolYear] = current_census_year
+          @homepage_data[:ncesSchoolId] = teachers_school.id
+          @homepage_data[:teacherName] = current_user.name
+          @homepage_data[:teacherId] = current_user.id
+          @homepage_data[:teacherEmail] = current_user.email
+        elsif donor_banner_name
+          teachers_school = current_user.last_complete_school_info.school
+
+          donor_teacher_banner_options = {}
+          donor_teacher_banner_options[:teacherFirstName] = current_user.short_name
+          donor_teacher_banner_options[:teacherSecondName] = current_user.second_name
+          donor_teacher_banner_options[:teacherEmail] = current_user.email
+          donor_teacher_banner_options[:ncesSchoolId] = teachers_school.id
+          donor_teacher_banner_options[:schoolAddress1] = teachers_school.address_line1
+          donor_teacher_banner_options[:schoolAddress2] = teachers_school.address_line2
+          donor_teacher_banner_options[:schoolAddress3] = teachers_school.address_line3
+          donor_teacher_banner_options[:schoolCity] = teachers_school.city
+          donor_teacher_banner_options[:schoolState] = teachers_school.state
+          donor_teacher_banner_options[:schoolZip] = teachers_school.zip
+
+          @homepage_data[:donorTeacherBannerOptions] = donor_teacher_banner_options
```

Comment:
This pattern of making a hash for the `donor_teacher_banner_options` seems to keep all of the related information in a tidy little bundle.  What about doing the same for the census banner?

Reply:
That's a good idea.  It worked for the donor teacher banner because the React classes (`TeacherHomepage` & `DonorTeacherBanner`) already expected that hash.  I don't want to modify JS here, just to keep scope limited, but it can go into a followup.